v7/tx3_tag_cloud/tx3_tag_cloud.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © 2015 Manuel Kaufmann
+
+# Permission is hereby granted, free of charge, to any
+# person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the
+# Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the
+# Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice
+# shall be included in all copies or substantial portions of
+# the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+# PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+# OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from __future__ import unicode_literals
+
+import os
+import json
+
+from nikola import utils
+from nikola.plugin_categories import Task
+
+
+class Tx3TagCloud(Task):
+
+    """Render the tags to be used by tx3TagCloud plugin."""
+
+    name = "tx3_tag_cloud"
+
+    def set_site(self, site):
+        """Set Nikola site."""
+        return super(Tx3TagCloud, self).set_site(site)
+
+    def gen_tasks(self):
+        """Render the tag pages and feeds."""
+        kw = {
+            "output_folder": self.site.config['OUTPUT_FOLDER'],
+            'tag_path': self.site.config['TAG_PATH'],
+            "tag_pages_are_indexes": self.site.config['TAG_PAGES_ARE_INDEXES'],
+            'category_path': self.site.config['CATEGORY_PATH'],
+            'category_prefix': self.site.config['CATEGORY_PREFIX'],
+            "taglist_minimum_post_count": self.site.config['TAGLIST_MINIMUM_POSTS'],
+            "pretty_urls": self.site.config['PRETTY_URLS'],
+            "strip_indexes": self.site.config['STRIP_INDEXES'],
+        }
+        self.site.scan_posts()
+
+        # Copy all the assets to the right places
+        asset_folder = os.path.join(os.path.dirname(__file__), "files")
+        for task in utils.copy_tree(asset_folder, kw["output_folder"]):
+            task["basename"] = str(self.name)
+            yield task
+
+        # Tag cloud json file
+        tag_cloud_data = {}
+        for tag, posts in self.site.posts_per_tag.items():
+            if tag in self.site.config['HIDDEN_TAGS'] or len(posts) < kw['taglist_minimum_post_count']:
+                continue
+            tag_cloud_data[tag] = [len(posts), self.site.link(
+                'tag', tag, self.site.config['DEFAULT_LANG'])]
+        output_name = os.path.join(kw['output_folder'],
+                                   'assets', 'js', 'tx3_tag_cloud.json')
```

Comment:
Because that task adds a *lot of information* that is not needed and make the render to be slow. Take a look at the difference in my blog site:
```
(blog)[12:45] humitos: nikola> ls -lah ./output/assets/js/tx3_tag_cloud.json ./output/assets/js/tag_cloud_data.json
-rw-rw-r-- 1 humitos humitos 279K ago 21 12:44 ./output/assets/js/tag_cloud_data.json
-rw-rw-r-- 1 humitos humitos 3,6K ago 20 22:11 ./output/assets/js/tx3_tag_cloud.json
(blog)[12:45] humitos: nikola>
```

3.6Kb against 279Kb. It's too much!